pkgs/applications/audio/sonata/default.nix

```diff
@@ -16,7 +16,7 @@ in buildPythonApplication rec {
 
   disabled = !isPy3k;
 
-  nativeBuildInputs = [ pkgconfig gettext ];
+  nativeBuildInputs = [ pkgconfig gettext glib ];
```

Comment:
this should be a `buildInputs` and a setup hook `wrapGAppsHook` should never be in `buildInputs`, but rather `nativeBuildInputs`.

Reply:
more details https://nixos.org/manual/nixpkgs/stable/#sec-language-gnome

Reply:
actually, the dependencies in here are very strange. `gobject-introspection` is propagated, but that  isn't a python dependency. `gobject-introspection` should be native  and `gtk3` should be build.
It also uses `pythonPath` and `propagatedBuildInputs`, and for a python package I don't see why we couldn't just use `pythonPath`. Also note that gobject-introspection's setup-hook isn't working correctly here https://github.com/NixOS/nixpkgs/issues/56943, so you need to set `strictDeps = false`. Also reference the issue in a comment above that line.

Reply:
`glib` is only used for the hooks here, that's why I put it in `nativeBuildInputs`.  I had no idea what was going on.  Thanks for pointing me to #56943, adding `strictDeps` worked!

Reply:
@gebner That issue was for `gobject-introspection` setup-hook not being run with strictDeps. This probably worked because `gtk3` propagates `glib`. I fixed up the things I noticed in 6de4b96228e97082803ec385af140ed8a859e48c